internal/pubsub/rabbitpubsub/rabbit.go

```diff
@@ -166,6 +139,54 @@ func (t *topic) SendBatch(ctx context.Context, ms []*driver.Message) error {
 	return <-errc
 }
 
+// Read from the channels established with NotifyPublish and NotifyReturn.
+// Must be called with t.mu held.
+func (t *topic) receiveFromPublishChannels(ctx context.Context, nMessages int) error {
+	// Consume all the acknowledgments for the messages we are publishing.
+	// Since SendBatch (the only caller of this method) holds the lock, we expect
+	// exactly as many acks as messages.
+	var err error
+	nAcks := 0
+	for nAcks < nMessages {
+		select {
+		case <-ctx.Done():
+			// Channel will be in a weird state (not all publish acks consumed, perhaps)
+			// so re-create it next time.
+			t.ch = nil
+			return ctx.Err()
+
+		case ret, ok := <-t.retc:
+			if !ok {
+				// Channel closed. Handled in the pubc case below. But set
+				// the channel to nil to prevent it from being selected again.
+				t.retc = nil
+			} else if err == nil {
+				// The message was returned from the server because it is unroutable.
+				// This will be the error we return, but continue so we drain all
+				// items from pubc. We don't need to re-establish the channel on this
+				// error.
+				err = fmt.Errorf("rabbitpubsub: message returned from %s: %s (code %d)",
```

Comment:
Note: eventually we probably will want to expose `ret` for `ErrorAs`. Is `ret` an error? If so, consider just returning it instead of creating a new error. If not, we may need to wrap it in an error.

Reply:
It's not an error.

Wait for the next PR before discussing error-handling. It is all about errors.